tools/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c

```diff
@@ -0,0 +1,23 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_IP_Private.h"
+#include "FreeRTOS_ARP.h"
+
+ARPPacket_t arp_packet;
+NetworkBufferDescriptor_t pxNetworkBuffer;
+
+//We assume that the pxGetNetworkBufferWithDescriptor function is implemented correctly and returns a valid data structure.
+//This is the mock to mimic the correct expected bahvior. If this allocation fails, this might invalidate the proof.
+NetworkBufferDescriptor_t *pxGetNetworkBufferWithDescriptor( size_t xRequestedSizeBytes, TickType_t xBlockTimeTicks ){
+	pxNetworkBuffer.pucEthernetBuffer = &arp_packet;
+	return &pxNetworkBuffer;
+}
```

Comment:
@markrtuttle please can you confirm that this is a good way to model this assumption?

Reply:
@karkhaz Malte is invoking cbmc with --nondet-static, so these structs are filled with unconstrained data.  But the function is returning the same pointer to the same struct on each invocation.  Assuming this function is called more than once, it might be better to malloc new structs on each invocation.

Reply:
I changed it to malloc a buffer of xRequestedSizeBytes.